packages/moonstone/Dropdown/Dropdown.js

```diff
@@ -133,7 +133,11 @@ const DropdownList = Skinnable(
 					select="radio"
 					selected={selected}
 				>
-					{children}
+					{children.map((o, i) => {
+						// eslint-disable-next-line no-shadow
+						const {children = o, key = i} = o;
```

Comment:
This is clever but as a result it might not be obvious what the purpose is. Some inline docs are warranted at least i think.

Reply:
I think I'd prefer this to use much less clever code, for easier readability:
```
{children.map((o, i) => 
	({children: o, key: o.key || `item${i}`, role: 'checkbox', 'aria-checked': selected === i})
```

Reply:
`0` is a valid key value for `o.key`, which would get replaced with `item${i}`

Reply:
@Djspaceg I've started exactly the same as you mentioned :) and then changed since I didn't want to make one line long but yeah.. readability is more important. Thanks~

Reply:
Feel free to place each prop on its own line. That's much easier to read too.
```
```javascript
{children.map((o, i) => ({
	children: (o.children || o),
	key: (o.key != null ? o.key : `item${i}`),
	role: 'checkbox',
	'aria-checked': (selected === i)
})
```